### PR TITLE
feat: unlock homeworld bgm and set homeworld bgm.

### PR DESCRIPTION
**Pull request by hamusuke0323**
## Description

Please carefully read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md) before making any pull requests.
And, **Do not make a pull request to merge into stable unless it is a hotfix. Use the development branch instead.**
## Type of changes

- [ ] Bug fix
- [x] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: src/main/java/emu/grasscutter/game/home/GameHome.java [hunk 1]
```diff
@@ -34,6 +35,7 @@ public class GameHome {
     int exp;
     List<FurnitureMakeSlotItem> furnitureMakeSlotItemList;
     ConcurrentHashMap<Integer, HomeSceneItem> sceneMap;
+    List<Integer> unlockedHomeBgmList;
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/game/home/GameHome.java]:
```suggestion
    Set<Integer> unlockedHomeBgmList;
```

**Comment by hamusuke0323** [comment]
Need to change packet names such as "Unk2700..."?

File: src/main/java/emu/grasscutter/game/systems/InventorySystem.java [hunk 2]
```diff
@@ -833,6 +833,17 @@ public GameItem useItem(Player player, long targetGuid, long itemGuid, int count
                     break;
                 }
                 break;
+            case MATERIAL_BGM:
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/game/systems/InventorySystem.java:836]: Does unlocking from the music player actually do a `UseItemReq`? Last time I tried, they weren't usable from inventory.

Comment (hamusuke0323) [reply src/main/java/emu/grasscutter/game/systems/InventorySystem.java:836]: > Does unlocking from the music player actually do a `UseItemReq`? Last time I tried, they weren't usable from inventory.

Yes. I saw `UseItemReq` in PacketSniffer of akebi

Comment (hamusuke0323) [reply src/main/java/emu/grasscutter/game/systems/InventorySystem.java:836]: image

After the confirm button was clicked, `UseItemReq` was appeared:
image

**Comment by Birdulon** [comment]
Sorry for the late review.

image
Can't seem to test on my end. :(

**Comment by Birdulon** [comment]
Ah, seems like it's just a softlock when there's no default spincrystals. Works fine manually adding some and restarting.

**Review by Birdulon — APPROVED** [review]

**Comment by Birdulon** [comment]
Ah, was on old resources set with different obfuscated names.

**Comment by eternalcomet** [comment]
can you help handle `<WARN:GadgetChest> Could not found the handler of this type of Chests SceneObj_Area_Common_Property_Ani_Prop_BGM_01` to enable player pick up homeworld bgm CDs(aka 旋曜玉帛）

**Pull request merged**